### ISSUE TITLE
[AS3] Flash Files with Box2d Engine

### ISSUE THREAD

**Issue by jhwgh1968** [ruffle-rs/ruffle#1848]
(I thought about putting this under the tracking issue, but since it was specific to a particular engine that might work on a subset of complete AS3, I decided to make a separate issue and let it be linked as blocking.)

I have a number of flash files which use [a port of the Box2D physics engine to AS3](https://www.box2dflash.org/about). The simplest one is their engine demo, which I managed to grab while it was still available.

If this file can be made to work, it should unlock quite a number of other files as well. Other flash player projects have also found it a helpful test case for an AS3 implementation.

[main_demo_21a.zip](https://github.com/ruffle-rs/ruffle/files/5646393/main_demo_21a.zip)

**Comment by jhwgh1968** [comment]
Blocked on #1368

**Comment by kmeisthax** [comment]
This will be a good test once we have working AS3 events and most of the existing display object types wired in. It doesn't appear to require much else.

**Comment by jhwgh1968** [comment]
A long overdue update.

Trying this file with master version `ee52ce32`, I am pleased to report it seems to work if I click through the AS3 warning. That said, the performance is significantly worse than the original flash -- where the library's entire point was to have good performance.

Based on this, I will leave it up to the team whether to close this or leave it open for further improvement.

**Comment by adrian17** [comment]
For the record, box2d is very much a good target for optimization and I regularly use it to test improvement PRs :)
I'm gonna keep it open till it's remotely usable at real-time.

**Comment by Lord-McSweeney** [comment]
Most box2d SWFs should be working now; closing in favor of #23227
